### PR TITLE
 Add PyCifStar to conda-forge 

### PR DESCRIPTION
**Pull request by MridulS**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by MridulS** [comment]
@conda-forge/help-python, ready for review!

Looks like rattler-build doesn't fully support https://github.com/conda-forge/cfep/blob/main/cfep-25.md as there isn't a way to add a lower pin for `tests.python` in the v1 recipe format. Should I just move to the old recipe format to make the linter happy?

File: recipes/pycifstar/recipe.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    - python ${{ python_min }}.*
+    - pip
+    - setuptools
+  run:
+    - python >=${{ python_min }}
+
+tests:
+  - python:
+      imports:
+        - pycifstar
```
Comment (beckermr) [inline recipes/pycifstar/recipe.yaml:30]:
```suggestion
        - pycifstar
  - script: python -c "import pycifstar"
    requirements:
      run:
        - python ${{ python_min }}
```

File: recipes/pycifstar/recipe.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  sha256: 5892fdf16c83372ee5f32557127d5f36e14b0bbe520883a4e2e70365382f70ed
+
+build:
+  number: 0
+  noarch: python
+  script: ${{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+
+requirements:
+  host:
+    - python ${{ python_min }}.*
```
Comment (beckermr) [inline recipes/pycifstar/recipe.yaml:21]:
```suggestion
    - python ${{ python_min }}
```

**Comment by beckermr** [comment]
@conda-forge-admin relint

**Comment by MridulS** [comment]
@conda-forge-admin relint

**Pull request merged**